### ISSUE TITLE
Leave some suggestions for future features

### ISSUE THREAD

**Issue by benjaminsabella** [zLeki/DxTrade-Api-Go#4]
Any ideas would be appreciated and taken into consideration in the future, please leave some below it would help me grow this project into something larger.

**Comment by potatoesforliferi** [comment]
personally i would like a simpler system of setting up things like Take profits and stop losses when i set a market order. 
I am going to try running your code on golang, when i get the time, but i couldnt see how to set up an order like

"market buy SL 500pips TP 1000pips" sort of input.

**Comment by kpathan** [comment]
Add feature to update stoploss and take profit

**Comment by benjaminsabella** [comment]
Gotcha, i'll work on it thank you for the suggestion

**Comment by zLeki** [comment]
Done - I have pushed it to main branch

**Comment by benjaminsabella** [comment]
> Add feature to update stoploss and take profit

Done - I have pushed it to main branch

**Comment by benjaminsabella** [comment]
> personally i would like a simpler system of setting up things like Take profits and stop losses when i set a market order. 
> 
> I am going to try running your code on golang, when i get the time, but i couldnt see how to set up an order like
> 
> 
> 
> "market buy SL 500pips TP 1000pips" sort of input. 

This has also been done and is pushed to the main branch

**Comment by potatoesforliferi** [comment]
awesome, saw the code changes to that TP SL thing zleki, just thinking the "examples" buy stuff needs to be updated now

**Comment by benjaminsabella** [comment]
> awesome, saw the code changes to that TP SL thing zleki, just thinking the "examples" buy stuff needs to be updated now

You're right, i'll edit them thanks.

**Comment by kpathan** [comment]
Would like to get market data

**Comment by benjaminsabella** [comment]
> Would like to get market data

Unfortunately it cannot be done due to their api restrictions

**Comment by kpathan** [comment]
is there anyway I can submit trailing stop loss with if then in order /modify order

**Comment by benjaminsabella** [comment]
> is there anyway I can submit trailing stop loss with if then in order /modify order

Yeah but calculations must be done server side because dxtrade doesn't offer trailing stops on their platform unfortunately

**Comment by potatoesforliferi** [comment]
i think an installation guide and running guide might be useful. Am currently noob at golang, so not knowing where to start is tough. :) 

Thank you for the library anyways :D

**Comment by patique** [comment]
@zLeki thank you very much for your awesome work

**Comment by 0xRowie** [comment]
@zLeki Thanks for your work, currently working on an EA for MT5 to copy trades to DXTrade, the only thing is that I cannot find any documentation using the https://dxtrade.*****.com/api/ endpoint? Where did you find this?

**Comment by benjaminsabella** [comment]
A lot of digging through the chrome network tab lmao @0xRowie

**Comment by 0xRowie** [comment]
> A lot of digging through the chrome network tab lmao @0xRowie

I was afraid you were going to say that, I'm doing so myself right now 😭 @zLeki 

Thanks for the quick response! I appreciate it.

**Comment by wrofe** [comment]
is there an ability to grab previous candle history?

**Comment by benjaminsabella** [comment]
To the best of my understanding, it seems like it can't be done. But, feel free to try your hand at it. I suggest you look at the websocket contents because the info you want is there. Sadly, the code can only get candlestick data in the browser.

**Comment by kpathan** [comment]
> is there an ability to grab previous candle history?

I was able to solve this using web socket and requesting for quote. Please check the network trace of the dxtrade. it has lot of features. 
I am not good at golang or python. so my code might not be useful

**Comment by benjaminsabella** [comment]
Any code will help, but thanks anyways this will be very helpful

**Comment by benjaminsabella** [comment]
@kpathan Thank you for your help! I am pushing an update which includes the ability to fetch candlestick data.

**Comment by kpathan** [comment]
I would like to calculate lot size with fixed amount to risk.

Ex: $250 to risk, With Entry, Stoploss. Take Profit is also know, but does not matter. 
I am new to forex, so I need some help with that

Thanks in Advance.

**Comment by kpathan** [comment]
> @kpathan Thank you for your help! I am pushing an update which includes the ability to fetch candlestick data.

Great work!!!

**Comment by mubasher24** [comment]
how can i run this, noob in go

**Comment by orfeysoft** [comment]
Hello, 

in the description of project is published the following information. 

"API for dxtrade that can be used with all golang applications to open, close, and manage positions with your dxtrade account. It works with all prop firms or companies that use dxtrade"

We have tried to connect this SDK with any DXTRADE account but without success. 
Could you provide some small example how to use your SDK with any DXTRADE account ? 

Thank you in advance.

**Comment by benjaminsabella** [comment]
> Hello,
> 
> in the description of project is published the following information.
> 
> "API for dxtrade that can be used with all golang applications to open, close, and manage positions with your dxtrade account. It works with all prop firms or companies that use dxtrade"
> 
> We have tried to connect this SDK with any DXTRADE account but without success. Could you provide some small example how to use your SDK with any DXTRADE account ?
> 
> Thank you in advance.

I cannot maintain this repository anymore since my contract with FTMO ended, we can troubleshoot if you would like in a new issue thread. Apologizes for the late reply.

**Comment by orfeysoft** [comment]
I have two different DXTrade Demo accounts and can provide them for testing. 
Please give me a valid email, I will send the account credentials. 
Thanks !

**Comment by benjaminsabella** [comment]
@orfeysoft shoot me an email at lekech.com@gmail.com and I'll give it a shot this week.
